tests/timeseries/test_periodic_feature_extractor.py

```diff
@@ -0,0 +1,215 @@
+    index_shifted = (
+        feature_extractor._timestamp_to_rel_index(timestamp)  # pylint: disable=W0212
+        % feature_extractor._distance  # pylint: disable=W0212
+    )
+    assert index_shifted == 1
+
+
+async def test_feature_extractor() -> None:
+    """Test the feature extractor with a moving window that contains data."""
+    start = UNIX_EPOCH + timedelta(seconds=1)
+    end = start + timedelta(seconds=2)
+
+    data: List[float] = [1, 2, 2.5, 1, 1, 1, 2, 2, 1, 1, 2, 2]
+
+    feature_extractor = await init_feature_extractor(data, timedelta(seconds=3))
+    assert np.allclose(feature_extractor.avg(start, end), [5 / 3, 4 / 3])
+
+    feature_extractor = await init_feature_extractor(data, timedelta(seconds=4))
+    assert np.allclose(feature_extractor.avg(start, end), [1, 2])
+
+    data: List[float] = [1, 2, 2.5, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1]  # type: ignore[no-redef]
+
+    feature_extractor = await init_feature_extractor(data, timedelta(seconds=5))
+    assert np.allclose(feature_extractor.avg(start, end), [1.5, 1.5])
+
+    async def _test_fun(data, period, start, end, expected, weights):
+        feature_extractor = await init_feature_extractor(data, timedelta(seconds=period))
+        ret = feature_extractor.avg(UNIX_EPOCH + timedelta(seconds=start), UNIX_EPOCH + timedelta(seconds=end), weights=weights)
+        print("{}, {}, {}, {}: {}".format(period, start, end, data[start:end], ret))
+        assert np.allclose(ret, expected)
+
+    async def test_09(period, start, end, expected, weights=None):
+        data: List[float] = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]  # type: ignore[no-redef]
+        await _test_fun(data, period, start, end, expected, weights)
+
+    async def test_011(period, start, end, expected, weights=None):
+        data: List[float] = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]  # type: ignore[no-redef]
+        await _test_fun(data, period, start, end, expected, weights)
+
+    print()
+    # empty time period
+    with pytest.raises(ValueError):
+        await test_09(period=5, start=1, end=1, expected=[4711])
+
+    # 3)
+    # moving window not multiple of period
+    with pytest.raises(ValueError):
+        await test_09(period=3, start=1, end=1, expected=[4711])
+
+    # 6)
+    # time period in moving window
+    await test_09(period=2, start=0, end=1, expected=[5])
+    #await test_09(period=2, start=0, end=2, expected=[5, 6]) # fails, should work
+    #await test_09(period=2, start=5, end=7, expected=[6, 5]) # fails, should work
+    #await test_09(period=2, start=8, end=10, expected=[5, 6]) # fails, should work
+    await test_09(period=5, start=0, end=1, expected=[5])
+    await test_09(period=5, start=5, end=6, expected=[5])
+
+    # 4+5)
+    # time period outside of moving window in future
+    await test_09(period=5, start=10, end=11, expected=[5])
+    # time period outside of moving window in past
+    await test_09(period=5, start=-5, end=-4, expected=[5, 6, 7, 8]) #should be [5]
```

Comment:
works now and probably was caused by a bug calculating the absolute value in `_timestamp_rel_index`